### PR TITLE
Fix bug in GUI when customizing a new plot

### PR DESCRIPTION
**Pull request by jim22k**
For the dataframe container, attempting to "customize" a new plot resulted in a KeyError inside `dfviz`. This fixes the issue.

Additionally, this PR allows the GUI to be opened for the non-builtin dataset prior to calling `intake.gui`.
This will now work:
```
import intake
ds = intake.open_catalog(...)
intake.interface.gui.GUI([ds])
```

### CODE REVIEW

File: intake/catalog/tests/test_catalog_save.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 def test_catalog_description():
```
Comment (martindurant) [inline intake/catalog/tests/test_catalog_save.py:10]: Was this test failing? It'd be good to make a test that would have caught the missing key in `__init__`.

Comment (jim22k) [reply intake/catalog/tests/test_catalog_save.py:10]: It wasn't failing, but was leaving the created test catalog in a non-temporary directory.

File: intake/catalog/tests/test_catalog_save.py [hunk 2]
```diff
@@ -10,6 +10,8 @@
...
     """Make sure the description comes through the save."""
 
+    NAME = 'desc_test.yaml'
```
Comment (martindurant) [inline intake/catalog/tests/test_catalog_save.py:13]: OK, got it - so this should be a tmpdir/tmpfile location

Comment (jim22k) [reply intake/catalog/tests/test_catalog_save.py]: Changed this to use tmpdir and removed the explicit delete.

**Pull request merged**